### PR TITLE
Modify savings goal panel

### LINKED ISSUE #192: Modify savings goal panel
Currently the data the savings goal panel displays include the following:
* Remaining budget
* Current savings
* Budget deficit
* Savings deficit

It is not useful to display all of the above information. Trivially, when the remaining budget is a positive value, the budget deficit will obviously be zero. Thus the savings goal panel should only display the budget and savings deficit only if these values are positive, and display the remaining budget and current savings otherwise.

### PR DESCRIPTION
**Pull request by yongping827**
Resolves #192.

Changes:
* Merge `remainingBudget` and `budgetDeficit` into `remainingBudget`
* Merge `currentSavings` and `savingsDeficit` into `currentSavings`
* Add condition to change text color of `remainingBudget` and `currentSavings` to red if deficit values are shown
* Modify 'SavingsGoalPanel to have a 'SavingsGoalCard' so that the 'SavingsGoalCard' does not occupy the entire right panel
* Add private constructor and factory method to `Amount` to handle negative values

### CODE REVIEW

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java [hunk 1]
```diff
@@ -43,13 +62,20 @@ public BigDecimal getValue() {
         return value;
     }
 
+    /**
+     * Returns true if the {@code Amount} has a non-negative value
```
Comment (ianyong) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java:66]:
```suggestion
     * Returns true if the {@code Amount} has a non-negative value.
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java [hunk 2]
```diff
@@ -28,6 +28,25 @@ public Amount(String amount) {
...
+     * @param value
+     */
+    private Amount(BigDecimal value) {
+        this.value = value;
+    }
+
+    /**
+     * Constructs an {@code Amount} from a BigDecimal value.
+     *
+     * @param value
```
Comment (ianyong) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java:43]: Missing description of param.

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java [hunk 3]
```diff
@@ -28,6 +28,25 @@ public Amount(String amount) {
         assert value.compareTo(BigDecimal.ZERO) >= 0; // amount should be non-negative
     }
 
+    /**
+     * Constructs an {@code Amount} from a BigDecimal value.
+     *
+     * @param value
```
Comment (ianyong) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java:34]: Missing description of param.

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/SavingsGoalCard.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    private Label monthlySavingsGoal;
+    @FXML
+    private Label budgetStatus;
+    @FXML
+    private Label savingsStatus;
+    @FXML
+    private ImageView savingsPicture;
+
+    /**
+     * Constructor of SavingsGoalCard.
```
Comment (ianyong) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/SavingsGoalCard.java:31]: Useless comment.

**Review by ianyong — CHANGES_REQUESTED** [review]

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/SavingsGoalCard.java (hunk 4 above)
Comment (yongping827) [reply src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/SavingsGoalCard.java:31]: Removing this comment fails checkstyle.

Comment (ianyong) [reply src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/SavingsGoalCard.java:31]: Try to make it more descriptive, I don't need a comment to tell me this is a constructor.

**Review by ianyong — APPROVED** [review]
LGTM!

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java [hunk 5]
```diff
@@ -43,13 +63,20 @@ public BigDecimal getValue() {
         return value;
     }
 
+    /**
+     * Returns true if the {@code Amount} has a non-negative value.
+     */
+    public boolean isNonNegative() {
+        return value.compareTo(BigDecimal.valueOf(0)) >= 0;
```
Comment (wltan) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java:70]:
```suggestion
        return value.compareTo(BigDecimal.ZERO) >= 0;
```

**Review by wltan — CHANGES_REQUESTED** [review]

**Review by wltan — DISMISSED** [review]
LGTM, using `Amount.of` might be breaking abstraction a little but we'll deal with that afterwards

**Review by wltan — APPROVED** [review]
LGTM https://github.com/AY2021S1-CS2103T-W16-3/tp/pull/207#pullrequestreview-519627068

**Review by wltan — APPROVED** [review]
Re-LGTM https://github.com/AY2021S1-CS2103T-W16-3/tp/pull/207#pullrequestreview-519639846

**Review by ianyong — APPROVED** [review]
LGTM!

**Pull request merged**